internal/server/iigointernal/judiciary.go

```diff
@@ -99,3 +115,125 @@ func (j *judiciary) appointNextPresident(currentPresident shared.ClientID, allIs
 	}
 	return nextPresident
 }
+
+// cycleSanctionCache rolls the sanction cahce one turn forward (effectively dropping any sanctions longer than the depth)
+func (j *judiciary) cycleSanctionCache() {
+	newMap := j.localSanctionCache
+	delete(newMap, sanctionCacheDepth-1)
+	newMapCache := newMap
+	for i := 0; i < sanctionCacheDepth-1; i++ {
+		newMap[i+1] = newMapCache[i]
+	}
+	newMap[0] = []roles.Sanction{}
+	j.localSanctionCache = newMap
+}
+
+// Helper functions //
+
+// getDefaultSanctionThresholds provides default thresholds for sanctions
+func getDefaultSanctionThresholds() map[roles.IIGOSanctionTier]roles.IIGOSanctionScore {
+	return map[roles.IIGOSanctionTier]roles.IIGOSanctionScore{
+		roles.SanctionTier1: 1,
+		roles.SanctionTier2: 5,
+		roles.SanctionTier3: 10,
+		roles.SanctionTier4: 20,
+		roles.SanctionTier5: 30,
+	}
+}
+
+// softMergeSanctionThresholds merges the default sanction thresholds with a (preferred) client version
+func softMergeSanctionThresholds(clientSanctionMap map[roles.IIGOSanctionTier]roles.IIGOSanctionScore) map[roles.IIGOSanctionTier]roles.IIGOSanctionScore {
+	defaultMap := getDefaultSanctionThresholds()
+	for k := range defaultMap {
+		if clientVal, ok := clientSanctionMap[k]; ok {
+			defaultMap[k] = clientVal
+		}
+	}
+	return defaultMap
+}
+
+// getIslandSanctionTier if statement based evaluator for which sanction tier a particular score is in
+func getIslandSanctionTier(islandScore roles.IIGOSanctionScore, scoreMap map[roles.IIGOSanctionTier]roles.IIGOSanctionScore) roles.IIGOSanctionTier {
+	if islandScore <= scoreMap[roles.SanctionTier1] {
```

Comment:
ok here. feel free to use `https://godoc.org/github.com/zenixls2/stl` especially lowerbound if you need more cases